packages/@aws-cdk/aws-apigateway/lib/restapi.ts

```diff
@@ -554,7 +561,7 @@ export abstract class RestApiBase extends Resource implements IRestApi {
     if (deploy) {
 
       this._latestDeployment = new Deployment(this, 'Deployment', {
-        description: 'Automatically created by the RestApi construct',
+        description: props.description? props.description :'Automatically created by the RestApi construct',
```

Comment:
Take a closer look at where this function is called. It's in two constructors, `SpecRestApi` and `RestApi`. The props for one of these doesn't have the `description` field. You'll need to add it there and then add some tests in both places for the output of `description` being set and not being set.

I will exempt a README change for this but it will require an update to the expected output of the integ tests as this will change the template.

Reply:
Thank you @TheRealAmazonKendra ! That raises another question I had - how do I update those integration test outputs? Do I have to run a CDK synth, and if so, what do I need to pass to it?

Reply:
See [INTEGRATION_TESTS.md](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)

Reply:
If `RestApiProps` has a description var, and 'SpecRestApiProps` extends `RestApiBaseProps`(where I added the var), do I need to add code to those constructors? Is it problematic that this description would possibly overlap with a description in the OpenAPI specification?

Reply:
I interpreted the both the error message and the code wrong. Scratch everything I said.

The issue here is that `RestApiProps` extends `RestApiOptions` which extends `RestApiBaseProps`. So, `description` is being declared twice in the context of `RestApiProps`. Since you are (correctly) adding it to `RestApiBaseProps` it should be removed from `RestApiProps`. I think that will solve the error.